tests/test_appsettings.py

```diff
@@ -426,165 +521,141 @@ def test_nested_dict_setting_required_both_inner_and_outer_setting(self):
+            settings=dict(a=appsettings.StringSetting(), b=appsettings.StringSetting(),), name="setting"
+        )
         setting.check()
-        assert setting.value == 123.456
-        assert type(setting.value) is float
+        assert setting.value["a"] == "A"
+        assert setting.value["b"] == "B"
 
-    @mock.patch.dict(os.environ, {"SETTING": "[1, 2, 3]"})
-    def test_iterable_setting_from_environ_json_value(self):
-        setting = appsettings.IterableSetting(name="setting")
-        setting.check()
-        assert setting.value == [1, 2, 3]
 
-    @mock.patch.dict(os.environ, {"SETTING": "1:2:3"})
-    def test_iterable_setting_from_environ_delimiter_value(self):
-        setting = appsettings.IterableSetting(name="setting")
-        setting.check()
-        assert setting.value == ["1", "2", "3"]
+class NestedSettingTestCase(SimpleTestCase):
+    """NestedSetting tests."""
 
-    @mock.patch.dict(os.environ, {"SETTING": "1-2-3"})
-    def test_iterable_setting_from_environ_delimiter_value_with_item_type(self):
-        setting = appsettings.IterableSetting(name="setting", item_type=int, delimiter="-")
-        setting.check()
-        assert setting.value == [1, 2, 3]
+    def test_nested_setting(self):
+        setting = appsettings.NestedDictSetting(settings=dict())
+        assert setting.value == {}
+        setting.transform_default = True
+        assert setting.value == {}
 
-    @mock.patch.dict(os.environ, {"SETTING": '"json-string"'})
-    def test_string_setting_from_environ_json_value(self):
-        setting = appsettings.StringSetting(name="setting")
-        setting.check()
-        assert setting.value == "json-string"
+        setting = appsettings.NestedDictSetting(
+            name="setting",
+            default={},
+            settings=dict(
+                bool1=appsettings.BooleanSetting(default=False),
+                bool2=appsettings.BooleanSetting(name="bool3", default=True),
+            ),
+        )
+        assert setting.value == {}
 
-    @mock.patch.dict(os.environ, {"SETTING": "pure-string"})
-    def test_string_setting_from_environ_pure_value(self):
-        setting = appsettings.StringSetting(name="setting")
-        setting.check()
-        assert setting.value == "pure-string"
+        with override_settings(SETTING={"BOOL3": False}):
+            assert setting.value == {"bool1": False, "bool2": False}
 
-    @mock.patch.dict(os.environ, {"SETTING": "a:b:b:b:c"})
-    def test_set_setting_from_environ_value(self):
-        setting = appsettings.SetSetting(name="setting")
-        setting.check()
-        assert setting.value == {"a", "b", "c"}
 
-    @mock.patch.dict(os.environ, {"SETTING": "a:b:c"})
-    def test_tuple_setting_from_environ_value(self):
-        setting = appsettings.TupleSetting(name="setting")
-        setting.check()
-        assert setting.value == ("a", "b", "c")
+class NestedListSetting(SimpleTestCase):
+    """NestedListSetting tests."""
```

Comment:
This test misses `TestCase` suffix, it should be `NestedListSettingTestCase`.

Reply:
Fixed.